lib/dor/item.rb

```diff
@@ -43,7 +43,7 @@ def sub_collections
         @sub_collections += members['sets'] if members['sets']
         @sub_collections += members['collections'] if members['collections']
       end
-      @sub_collections
+      @sub_collections.delete_if {|collection| collection['druid'] == druid} # if this is a collection, do not include yourself in the sub-collection list
```

Comment:
Shouldn't the `delete_if` statement live inside the `unless @sub_collections` block? (Otherwise, you're presumably unnecessarily running the `delete_if` every time `sub_collections` is called).